inspirehep/modules/workflows/views/callback.py

```diff
@@ -153,12 +179,79 @@ def webcoll_callback():
         )
         db.session.delete(pending_record)
         db.session.commit()
-        if response[recid].get('status') == 'failed':
-            continue
 
     return jsonify(response)
 
 
+def _robotupload_has_error(result):
+    recid = int(result.get('recid'))
```

Comment:
It should always have a recid according to the interface definition, if not, then it should fail yep.